### DISCUSSION TITLE
Co-pilot

### DISCUSSION THREAD

**Discussion by krosenfeld-IDM** [InstituteforDiseaseModeling/laser#21] [General]
@jonathanhhb 's [built a RAG style chatbot](https://github.com/InstituteforDiseaseModeling/laser/wiki/Towards-a-LASER-Co%E2%80%90Pilot) for the source code.

I think this is a good idea. Is there a way to incorporate feedback into code that someone is then writing? So not just asking questions but something that is helping you build. I'm thinking about GitHub Copilot where it can make suggestions based off of e.g. comments or instructions. Imagine:
```
demog = laser.basic_pop()
# Initialize a population of 1 million people with 100 randomly connected nodes using the pre-loaded demographics
```
and then it gives you the initialization?

**Comment by krosenfeld-IDM** [comment]
I also would like "guide star" to be in consideration for any naming (see [laser guide star](https://en.wikipedia.org/wiki/Laser_guide_star))

**Comment by clorton** [comment]
I think I would like `basic_pop()` to have some optional parameters such as the total population size (default=10^6 in your example) and number of nodes (default=100 in your example).

@jonathanhhb  Also, what do you mean by "the pre-loaded demographics"?

**Reply by jonathanhhb** [reply]
Sorry, where is that quoted from?

**Reply by krosenfeld-IDM** [reply]
I think @clorton was referring to my imaginary copilot instruction. I think the term "pre-loaded" was confusing - hopefully this version is clearer:
```
# load demographics for a basic population containing the initial age distribution, age based mortality, and birth rates.
demog = laser.basic_pop()
# Initialize a population of 1 million people with 100 randomly connected nodes using the demographics contained in `demog`
<copilot assist>
```
